hazelcast-client-new/src/test/java/com/hazelcast/client/map/ClientMapTest.java

```diff
@@ -226,6 +226,20 @@ public void testGetAllPutAll() {
     }
 
     @Test
+    public void testPutAllWithTooManyEntries() {
+        final IMap map = createMap();
+        Map mm = new HashMap();
+        for (int i = 0; i < 1000; i++) {
```

Comment:
This was originally intended to exercise the case where the number of entries exceeded partitionCount * 3, which if so, would execute the group-by-partition logic you see in this PR; otherwise, it would iterate through the entries sending each out as a PutOperation. 1000 was > 3 * 271, the default partition count in Hazelcast, which I wasn't able to immediately work out how to extract.